### PR TITLE
[KYUUBI #3104] Support SSL for Etcd

### PR DESCRIPTION
**Pull request by hddong**
### _Why are the changes needed?_

Support SSL for Etcd

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala [hunk 1]
```diff
@@ -191,4 +191,32 @@ object HighAvailabilityConf {
       .timeConf
       .checkValue(_ > 0, "Must be positive")
       .createWithDefault(Duration.ofSeconds(10).toMillis)
+
+  val HA_ETCD_SSL_ENABLE: ConfigEntry[Boolean] =
+    buildConf("kyuubi.ha.etcd.ssl.enable")
```
Comment (pan3793) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala:196]: enable => enabled

Comment (hddong) [reply kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala]: > enable => enabled

Done!

**Comment by hddong** [comment]
@pan3793 please review this again when free.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**